pkg/ingester/ingester_v2.go

```diff
@@ -121,9 +125,33 @@ func NewV2(cfg Config, clientConfig client.Config, limits *validation.Overrides,
 		go i.shipBlocksLoop()
 	}
 
+	i.done.Add(1)
+	go i.rateUpdateLoop()
+
 	return i, nil
 }
 
+func (i *Ingester) rateUpdateLoop() {
+	defer i.done.Done()
+
+	rateUpdateTicker := time.NewTicker(i.cfg.RateUpdatePeriod)
+	defer rateUpdateTicker.Stop()
+
+	for {
+		select {
+		case <-rateUpdateTicker.C:
+			i.userStatesMtx.RLock()
+			for _, state := range i.TSDBState.dbs {
```

Comment:
I would rename `state` into `db` for consistency with the rest of the code base.

Reply:
OK. i will rename that.